---
title: "Phosphoproteomics - Project 4"
output: html_notebook
---


```{r}
# Load packages

if (!require("pacman")) install.packages("pacman")
pacman::p_load(dplyr, tidyr, ggplot2, BiocManager, ggrepel)

library(dplyr)
library(tidyr)
library(ggplot2)
library(ggrepel)

BiocManager::install("STRINGdb")
library(STRINGdb)
```


```{r}
# Read in data
phospho <- read.csv("/home/noah/phosphoproteomics_project/phosphoproteomics/Dataset_Phospho.csv", header = T, stringsAsFactors = F, sep = ";")
proteome <- read.csv("phosphoproteomics/Dataset_Proteome.csv", header = T, stringsAsFactors = F, sep = ";")
dim(phospho)

# Remove rows with NA in any of the abundance columns
phospho <- phospho %>% drop_na(Abundance.Control.1, Abundance.Control.2, Abundance.Control.3, Abundance.LIF.1, Abundance.LIF.2, Abundance.LIF.3)
dim(phospho)

# Average of abundance columns for control and LIF
phospho$Control_avg <- rowMeans(phospho[,c(10, 11, 12)], na.rm = T)
phospho$LIF_avg <- rowMeans(phospho[,c(13, 14, 15)], na.rm = T)

# Calculate LIF/Control ratio
phospho$FC <- phospho$LIF_avg / phospho$Control_avg

```


# Histogram for ratios
```{r}
# Histogram of ratios

# PLOT FC (phospho dataaset)
lower = quantile(phospho$FC, probs = .25)
middle = quantile(phospho$FC, probs = .50)
upper = quantile(phospho$FC, probs = .75)

ggplot(phospho, aes(x=FC)) + 
  geom_histogram(binwidth=0.005) +
  geom_vline(xintercept = lower, color = "brown2") +
  geom_vline(xintercept = middle) +
  geom_vline(xintercept = upper, color = "brown2") +
  geom_text(aes(x=lower-0.3, label="0.899 (25%)", y=40), color = "brown2", text=element_text(size=8)) +
  geom_text(aes(x=upper+0.3, label="1.123 (75%)", y=40), color = "brown2", text=element_text(size=8)) +
  xlim(0, 3)+
  xlab("FC")+
  ylab("Count")+
  ggtitle("Distribution FC (LIF/Control)")


# PLOT log2FC (phospho dataset)
log2FC <- log2(phospho$FC)

log2FC_lower = quantile(log2FC, probs = .25)
log2FC_middle = quantile(log2FC, probs = .50)
log2FC_upper = quantile(log2FC, probs = .75)

ggplot(phospho, aes(x=log2FC)) + 
  geom_histogram(binwidth=0.005) +
  geom_vline(xintercept = log2FC_lower, color = "brown2") +
  geom_vline(xintercept = log2FC_middle) +
  geom_vline(xintercept = log2FC_upper, color = "brown2") +
  geom_text(aes(x=log2FC_lower-0.3, label="-0.155 (25%)", y=40), color = "brown2", text=element_text(size=8)) +
  geom_text(aes(x=log2FC_upper+0.3, label="1.169 (75%)", y=40), color = "brown2", text=element_text(size=8)) +
  xlim(-2, 2)+
  xlab("log2FC")+
  ylab("Count")+
  ggtitle("Distribution log2FC")


```



# p-values
```{r}
# ALT 2
df_copy <- tibble(phospho) # Copy df
df_copy[] <- lapply(df_copy, function(x) as.numeric(as.character(x))) 
phospho$pValues <- apply(df_copy, 1, function(x) t.test(x[10:12],x[13:15], alternative = "less", paired = T)$p.value)

# ADjusted p-values?
```


# VOLCANO PLOT
```{r}

# Volcano plot
  # Use pval (instead of adjusted to separate)
  # Threshold ratio threshold (red vertical) ratio 2 -> log2(2) -> 1
  # Threshold pval -> 0.05

### Proteome volcano plot
proteome$log2_FC <- log2(proteome$Abundance.Ratio...LIF.....Control.)
proteome$diffexpressed <- "NO"
proteome$diffexpressed[proteome$log2_FC > 1 & proteome$Abundance.Ratio.P.Value...LIF.....Control. < 0.05] <- "UP"
proteome$diffexpressed[proteome$log2_FC < -1 & proteome$Abundance.Ratio.P.Value...LIF.....Control. < 0.05] <- "DOWN"

ggplot(data=proteome, aes(x=log2_FC, y=-log10(Abundance.Ratio.P.Value...LIF.....Control.), col=diffexpressed))+
  geom_point()+
  theme_minimal()+
  geom_vline(xintercept=c(-1, 1), col="red") +
  geom_hline(yintercept=-log10(0.05), col="red")


### Phospho volcano plot
phospho$log2_FC <- log2(phospho$FC)
phospho$diffexpressed <- "NO"
phospho$diffexpressed[phospho$log2_FC > 1 & phospho$pValues < 0.05] <- "UP"
phospho$diffexpressed[phospho$log2_FC < -1 & phospho$pValues < 0.05] <- "DOWN"

phospho$delabel <- NA
phospho$delabel[phospho$diffexpressed != "NO"] <- phospho$Master.Protein.Accessions[phospho$diffexpressed != "NO"]
phospho$delabel

##Addition to Alices code to generate "real names" not Uniprot IDs
Mapping_phospho <- string_db$map(phospho, "delabel", removeUnmappedRows = F)
Mapping_phospho <- string_db$add_proteins_description(Mapping_phospho) #Add´s protein information to STRINGid´s 
colnames(Mapping_phospho)

#Volcano plot overview
ggplot(data=Mapping_phospho, aes(x=log2_FC, y=-log10(pValues), col=diffexpressed, label=preferred_name))+
  geom_point()+
  theme_minimal()+
  geom_vline(xintercept=c(-1, 1), col="red") +
  geom_hline(yintercept=-log10(0.05), col="red")

ggsave("~/phosphoproteomics_project/results/Zoomed_out_Volcano.png", width = 1500, height = 1000, units = "px")
#Detail/Zoomed in
ggplot(data=Mapping_phospho, aes(x=log2_FC, y=-log10(pValues), col=diffexpressed, label=preferred_name))+
  geom_point()+
  theme_minimal()+
  geom_vline(xintercept=c(-1, 1), col="red") +
  geom_hline(yintercept=-log10(0.05), col="red") + 
  coord_cartesian(xlim = c(1,3.5), ylim = c(1.25, 4.5))+
  geom_label_repel()

ggsave("~/phosphoproteomics_project/results/Zoomed_in_Volcano.png", width = 1500, height = 1000, units = "px")

```



# Filter function
```{r}

filter_proteins <- function(df_phospho, df_proteome, phospho_type,proteome_type, phospho_ratio = 2, lower = 0.9, upper = 1.1) {
  
  ## Filter PHOSPHO
  
  # 1 -> Ratio and pval
  if (phospho_type == "ratio"){ 
    filtered_phospho <- phospho[which(df_phospho$pValues<=0.05 & df_phospho$FC>=phospho_ratio),] 
  }
  
  # 2 -> Adj_pval
  if (phospho_type == "phospho_pval_adj"){
    filtered_phospho <- phospho[which(df_phospho$PVal_adj<=0.009),]
  }
  
  # Filter PROTEOME
  
  # 1 -> Keep non-significant
  if (proteome_type == "non_singificant"){
    filtered_proteome <- df_proteome[which(df_proteome$Abundance.Ratio.P.Value...LIF.....Control.>=0.05),] 
  }

  # 2 -> Keep non-significant (pval > 0.05) within range 
  if (proteome_type == "range"){
    filtered_proteome <- df_proteome[which(between(df_proteome$Abundance.Ratio...LIF.....Control., lower, upper) &
                                            df_proteome$Abundance.Ratio.P.Value...LIF.....Control.>=0.05),] 
  }
  
  # 3 -> pVal adjusted
  if (proteome_type == "proteome_pval_adj"){
    filtered_proteome <- df_proteome[which(df_proteome$Abundance.Ratio.Adj..P.Value...LIF.....Control.>=0.05),] 
  }
  
  # Combine phospho and proteome filter
  More_Phosphorylated <- merge(filtered_phospho, filtered_proteome, by.x="Master.Protein.Accessions", by.y = "Accession", all = F)
  
  return(More_Phosphorylated) 
  
  
}


# phospho_type, specify: 
  # "ratio" - pval <= 0.05, need to specify ratio
  # "phospho_pval_adj" - pval <= 0.05/6 (Bonferroni)
# proteome_type, specify:
  # "non_singificant" - keep pval >= 0.05 (non-significant)
  # "range" - pval >= 0.05, need to specify range
  # "proteome_pval_adj" - keep pval_adj >= 0.05 (non-significant)

More_Phosphorylated <- filter_proteins(phospho, proteome, phospho_type = "ratio", proteome_type = "proteome_pval_adj", phospho_ratio = 2, lower = 0.9, upper = 1.1)

dim(phospho)
dim(proteome)
dim(More_Phosphorylated)


```


# Stringdb
```{r}
# Initialization
# Score_threshold = threshold for combined scores/evidence of interactions
string_db <- STRINGdb$new(version="11.0b", species=10090, score_threshold = 200, input_directory="")

# Map gene names to STRING database identifiers (add column of string-id)
hits <- string_db$map(More_Phosphorylated, "Master.Protein.Accessions", removeUnmappedRows = T)

# Plot network
string_db$plot_network(hits$STRING_id)

# Save png
string_db$get_png(hits$STRING_id, file = "/Users/alice/Bioinformatics_course/Project4/plot.png")

```


# Plot Stat5b subnetwork
```{r}
# Get interactions  
interactions <- string_db$get_interactions(hits$STRING_id)

# Get string id of Stat5b gene using mp method
Stat5b <- string_db$mp("Stat5b")

# Grep Stat5b id from interactions columns
Stat5b_from_to <- interactions[Reduce(`|`, lapply(interactions, grepl, pattern = Stat5b)),]

## Plot interactions from & to Stat5b
Stat5b_all <- unlist(Stat5b_from_to[1:2])       # Make one list of "from" & "to" column
Stat5b_all <- as.data.frame(Stat5b_all)         # Make list into data frame
string_db$plot_network(Stat5b_all)              # Plot all Stat5b interactions

## Plot interactions from Stat5b
string_db$plot_network(Stat5b_from_to$from)

```



# More info
```{r}

# Enrichment
enrichment <- string_db$get_enrichment(Stat5b_from_to, category = "KEGG")

eee <- string_db$get_ppi_enrichment(interactions)

# Protein descriptions
colnames(Stat5b_all) <- c("STRING_id")
protein_descriptions <- string_db$add_proteins_description(Stat5b_all)
```


